src/drawer/drawer.tsx

```diff
@@ -52,6 +64,36 @@ export const Drawer = ({
         event.stopPropagation();
     };
 
+    const handleTabKeyPress = (e: KeyboardEvent) => {
```

Comment:
suggest to use the Floating UI's `FloatingFocusManager` to perform focus trapping instead, tab detection is not sufficient for Android Talkback

you may refer to the latest changes for the Filter component: https://github.com/LifeSG/react-design-system/pull/788/files